src/runtime_src/core/pcie/driver/linux/xocl/subdev/mailbox.c

```diff
@@ -1210,7 +1270,21 @@ static bool do_hw_rx(struct mailbox_channel *ch)
 		return false;
 	}
 
-	chan_recv_pkt(ch);
+	/*
+	 * Don't trust the peer. If we think the peer is doing something
+	 * malicious, we disable interrupt and don't handle the pkts. Once
+	 * this happened, user can't use mailbox anymore before admin manually
+	 * recovers the mailbox by doing 'xbmgmt reset --hot --card xxx'
+	 * This is the protection at the pkt layer. The malicious user can
+	 * still escape the protection here by carefully control the sending
+	 * pkt rate. At msg layer, we have another type of protection -- we
+	 * discard those msg requests which are disabled by admin. eg,
+	 * xclbin download (type 0x8) is not expected to show up on h/w mailbox
```

Comment:
isn't the channel switch good enough for this? if you don't want to see 0x8 on hw channel, you can redirect it to sw channel (assuming you already have some kind of protection in place for sw channel).

Reply:
2 per channel variables required